server/aws-lsp-codewhisperer/src/language-server/telemetryService.ts

```diff
@@ -0,0 +1,97 @@
+import { CodeWhispererServiceToken } from './codeWhispererService'
+import { CredentialsProvider, Telemetry } from '@aws/language-server-runtimes/server-interface'
+import { CodeWhispererSession } from './session/sessionManager'
+import {
+    SuggestionState,
+    UserTriggerDecisionEvent,
+    UserContext,
+    OptOutPreference,
+} from '../client/token/codewhispererbearertokenclient'
+import { getCompletionType } from './utils'
+
+export class TelemetryService extends CodeWhispererServiceToken {
+    private userContext!: UserContext
+    private optOutPreference!: OptOutPreference
+    private enableTelemetryEventsToDestination!: boolean
+    private telemetry: Telemetry
+
+    constructor(
+        credentialsProvider: CredentialsProvider,
+        additionalAwsConfig: AWS.ConfigurationOptions = {},
+        telemetry: Telemetry
+    ) {
+        super(credentialsProvider, additionalAwsConfig)
+        this.telemetry = telemetry
+    }
+
+    public updateUserContext(userContext: UserContext): void {
+        this.userContext = userContext
+    }
+
+    public updateOptOutPreference(optOutPreference: OptOutPreference): void {
+        this.optOutPreference = optOutPreference
+    }
+
+    public updateEnableTelemetryEventsToDestination(enableTelemetryEventsToDestination: boolean): void {
+        this.enableTelemetryEventsToDestination = enableTelemetryEventsToDestination
+    }
+
+    private getSuggestionState(session: CodeWhispererSession): SuggestionState {
+        let suggestionState: SuggestionState
+        if (session.getAggregatedUserTriggerDecision() === undefined) {
```

Comment:
(super nit pick and personal choice) why not using `switch-case` here? Makes code much easier to read

Reply:
you are right, switch-case looks a bit cleaner, will update